### DISCUSSION TITLE
Execution ina standalone Python script with PyQt

### DISCUSSION THREAD

**Discussion by sharoseali** [opengeos/segment-geospatial#282] [Q&A]
HI @giswqs  Thank you for your great work. I have successfully run the code on Colab. I want to implement in python script with Qt or pyqt. I have installed all the dependencies in conda Environment, including the segment-geospatial library . Do you have any information or suggestions on how to replicate these map interactive widgets using a PyQt script?

Thanks

**Comment by giswqs** [comment]
Does PyQt support ipywidgets?

**Reply by sharoseali** [reply]
Well, it is possible to do something using  [QtViola](https://github.com/Tauffer-Consulting/qtvoila) This tool runs the notebook cells and displays cell output in the qt window inside QtwebWidget. Also, there is one library called pyqtlet. This can display a normal street map inside the QT window but I am not sure how to display the Sattelite base and interactive map using it. ON THE LEAFMAP website. It mentions that Leafmap is built on leaflet. Whereas there is one open-source library called [pyqtlet](https://github.com/skylarkdrones/pyqtlet). It is a wrapper for Leaflet maps in PyQt5. To display open street map. the code works like this:
import sys
from PyQt5.QtWidgets import QApplication, QVBoxLayout, QWidget
from pyqtlet import L, MapWidget

```

class MapWindow(QWidget):
    def __init__(self):
        # Setting up the widgets and layout
        super().__init__()
        self.mapWidget = MapWidget()
        self.layout = QVBoxLayout()
        self.layout.addWidget(self.mapWidget)
        self.setLayout(self.layout)

        # Working with the maps with pyqtlet
        self.map = L.map(self.mapWidget)
        self.map.setView([12.97, 77.59], 10)
        L.tileLayer('http://{s}.tile.osm.org/{z}/{x}/{y}.png').addTo(self.map)
        self.marker = L.marker([12.934056, 77.610029])
        self.marker.bindPopup('Maps are a treasure.')
        self.map.addLayer(self.marker)
        self.show()

if __name__ == '__main__':
    app = QApplication(sys.argv)
    widget = MapWindow()
    sys.exit(app.exec_())
```
It seems doing some transtive work (using a stack of two or more libs ) can create a standalone desktop application for SamGeo.
